### PR TITLE
Unit testing support added with jest and react testing library

### PR DESCRIPTION
**Pull request by mdshamoon**
## Summary

Hey guys, keep up the good work.
I can see that there is an issue with adding integration tests using cypress. I would like to help with that.
Also, I believe that some unit testing should also be done for avoiding minor bugs in the system. So I have added support for jest and react-testing library in this PR.

You can run the test cases using the command `yarn test`. Before that, you need to install the dependencies.

Currently, I have added a test case for a single component named `H1`. If you guys want, I can add more.

Note: You can read about unit testing in gatsby here- https://gatsby.dev/unit-testing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,26 +11,36 @@
...
 		"@types/react-select": "^3.0.22",
+		"@types/react-test-renderer": "^17.0.0",
+		"babel-jest": "^26.6.3",
+		"babel-preset-gatsby": "^0.7.0",
 		"classnames": "^2.2.6",
 		"expo": "^37.0.12",
 		"gatsby": "^2.25.3",
 		"i18n-js": "^3.8.0",
+		"identity-obj-proxy": "^3.0.0",
+		"jest": "^26.6.3",
```
Comment (amaurymartiny) [inline package.json:37]: could you add all those dependencies in devDependencies?

**Review by amaurymartiny — COMMENTED** [review]
Nice 🎉 

> Currently, I have added a test case for a single component named H1. If you guys want, I can add more.

That would be nice, but in subsequent PRs. Let's keep this PR smallish.

Could you add a github job in `pr.yml` that runs `yarn test`? If that job is green, I'll merge 👍

File: package.json (hunk 1 above)
Comment (mdshamoon) [reply package.json:37]: Sure.

File: jest-preprocess.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const babelOptions = {
```
Comment (amaurymartiny) [inline jest-preprocess.js:1]: could we move this file into a new folder `src/testutil/jest-preprocess.js`

I prefer not to have too many files on the root folder

File: loadershim.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+global.___loader = {
```
Comment (amaurymartiny) [inline loadershim.js:1]: same

File: setupTests.js [hunk 4]
```diff
@@ -0,0 +1 @@
+import '@testing-library/jest-dom/extend-expect';
```
Comment (amaurymartiny) [inline setupTests.js:1]: same.

I think we can use `src/testutil/setupTest.ts` (.ts file) for this one.

File: jest.config.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
+	transform: {
+		'^.+\\.[jt]sx?$': '<rootDir>/jest-preprocess.js',
```
Comment (amaurymartiny) [inline jest.config.js:3]:
```suggestion
		'^.+\\.[jt]sx?$': '<rootDir>/src/testutil/jest-preprocess.js',
```

File: jest.config.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	],
+	transformIgnorePatterns: [
+		`node_modules/(?!(gatsby|react-native|expo-constants|@unimodules|@expo|expo-font)/)`,
+	],
+	globals: {
+		__PATH_PREFIX__: ``,
+		__DEV__: true,
+	},
+	testURL: `http://localhost`,
+	setupFiles: [`<rootDir>/loadershim.js`],
```
Comment (amaurymartiny) [inline jest.config.js:23]:
```suggestion
	setupFiles: [`<rootDir>/src/testutil/loadershim.js`],
```

File: jest.config.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+	setupFilesAfterEnv: ['<rootDir>/setupTests.js'],
```
Comment (amaurymartiny) [inline jest.config.js:24]:
```suggestion
	setupFilesAfterEnv: ['<rootDir>/src/testutil/setupTests.js'],
```

**Review by amaurymartiny — COMMENTED** [review]
LGTM! Final comments about moving some files.

Could you also solve the linting issues?

1. You can remove those 3 lines (they are not really needed):
https://github.com/shootismoke/webapp/blob/e6338d439a495e61e2a48c4bf8d566cdcee5dc64/cypress/integration/components/SearchBar.ts#L26-L29

2. Maybe https://spectrum.chat/testing-library/help/error-when-using-typescript-tobeinthedocument~2d6943a9-b67d-40c4-9c53-3a4db9bcfd46 is relevant for the 2nd lint error?

**Review by amaurymartiny — APPROVED** [review]
Thanks a lot!

**Pull request merged**